src/compiler/scala/tools/nsc/tasty/bridge/TypeOps.scala

```diff
@@ -187,31 +183,25 @@ trait TypeOps extends TastyKernel { self: TastyUniverse =>
 
     /**Best effort to transform this to an equivalent canonical representation in scalac.
      */
-    final def canonicalForm: Type = {
-      val resUpper = resType.upperBound
-      val resLower = if (resType `eq` resType.bounds) resType.lowerBound else defn.NothingTpe
-      if (resUpper.typeArgs.nonEmpty && resUpper.typeArgs == paramInfos) {
-        val resUpperRef = resUpper.asInstanceOf[TypeRef]
-        mkPolyType(
-          typeParams,
-          mkTypeBounds(
-            resLower,
-            mkExistentialType(
-              typeParams,
-              mkTypeRef(resUpperRef.pre, resUpperRef.sym, typeParams.map(_.tpe))
-            )
-          )
-        )
+    final def canonicalForm(implicit ctx: Context): Type = {
+      ctx.log(s"canonical form of $this")
+      def eliminateBadArgs(tpe: Type): Type = {
+        if (tpe.typeArgs.exists(_ =:= TypeBounds.empty))
```

Comment:
this is supposed to overcome what dotty encodes for `[F[_] <: List[_]]` but probably would fail for `L <: List[_ <: Int]`

Reply:
new implementation peels off hk params from bounds with hk params in either side